tests/gui/modules/progress-status.spec.js

```diff
@@ -0,0 +1,86 @@
+'use strict';
+
+const m = require('mochainon');
+const settings = require('../../../lib/gui/models/settings');
+const progressStatus = require('../../../lib/gui/modules/progress-status');
+
+describe('Browser: progressStatus', function() {
+
+  describe('.fromFlashState()', function() {
+
+    beforeEach(function() {
+      this.state = {
+        type: 'write',
+        percentage: 0,
+        eta: 15,
+        speed: 100000000000000
+      };
+
+      settings.set('unmountOnSuccess', true);
+    });
+
+    it('should report 0% if percentage == 0 but speed != 0', function() {
+      m.chai.expect(progressStatus.fromFlashState(this.state)).to.equal('0% Flashing');
+    });
+
+    it('should handle percentage == 0, type = write, unmountOnSuccess', function() {
+      this.state.speed = 0;
+      m.chai.expect(progressStatus.fromFlashState(this.state)).to.equal('Starting...');
+    });
+
+    it('should handle percentage == 0, type = write, !unmountOnSuccess', function() {
+      this.state.speed = 0;
+      settings.set('unmountOnSuccess', false);
+      m.chai.expect(progressStatus.fromFlashState(this.state)).to.equal('Starting...');
+    });
+
+    it('should handle percentage == 0, type = check, unmountOnSuccess', function() {
+      this.state.speed = 0;
+      this.state.type = 'check';
+      settings.set('unmountOnSuccess', false);
+      m.chai.expect(progressStatus.fromFlashState(this.state)).to.equal('Starting...');
```

Comment:
Hmmm, looks weird to have the text reset back to `Starting...` when `type == check` ? :-/
Perhaps in this case it should just say `Validating...`? What do you think @jviotti ?